do_everything.sh

```diff
@@ -123,166 +114,7 @@ echo
-    # apply_patch $my_loc/patches/laser_filters.patch
-
-    # Patch camera_info_manager - remove testing for Android
-    # TODO: It seems like there may be a better way to handle the test issues
-    # http://stackoverflow.com/questions/22055741/googletest-for-android-ndk
-    # https://source.android.com/reference/com/android/tradefed/testtype/GTest.html
-    apply_patch $my_loc/patches/camera_info_manager.patch
-
-    # Patch camera_calibration_parsers - deleted python things and solved problem finding Boost.
-    apply_patch $my_loc/patches/camera_calibration_parsers.patch
-
-    # Patch cv_bridge - fix transitive linking in cv_bridge-extras.cmake
-    apply_patch $my_loc/patches/cv_bridge.patch
-
-    # Patch theora_image_transport - fix transitive linking
-    apply_patch $my_loc/patches/theora_image_transport.patch
-
-    # Patch robot_pose_ekf - Add bfl library cmake variables, also, remove tests
-    # TODO: The correct way to handle this would be to create .cmake files for bfl and do a findpackage(orocos-bfl)
-    # apply_patch $my_loc/patches/robot_pose_ekf.patch
-
-    # Patch robot_state_publisher - Add ARCHIVE DESTINATION
-    # TODO: Create PR to add ARCHIVE DESTINATION
-    apply_patch $my_loc/patches/robot_state_publisher.patch
-
-    # Patch moveit_core - Add fcl library cmake variables
-    # TODO: The correct way to handle this would be to create .cmake files for fcl and do a findpackage(fcl)
-    #apply_patch $my_loc/patches/moveit_core.patch
-
-    # Patch moveit_core plugins - Add ARCHIVE DESTINATION
-    # TODO: PR merged: https://github.com/ros-planning/moveit_core/pull/251
-    # Wait for next release to remove (current 0.6.15)
-    #apply_patch $my_loc/patches/moveit_core_plugins.patch
-
-    # Patch camera_calibration_parsers - Fix yaml-cpp dependency
-    # TODO: PR created: https://github.com/ros-perception/image_common/pull/36
-    # apply_patch $my_loc/patches/camera_calibration_parsers.patch
-
-    # Patch image_view - Solved YAML linking problems, and transitive linking.
-    apply_patch $my_loc/patches/image_view.patch
-
-    # Patch depth_image_proc - Solved transitive linking problems
-    apply_patch $my_loc/patches/depth_image_proc.patch
-
-    # Patch urdf - Fixed linking with pluginlib and dependencies in downstream packages
-    apply_patch $my_loc/patches/urdf.patch
-
-    # Patch global_planner - Add angles dependency
-    # TODO: PR merged: https://github.com/ros-planning/navigation/pull/359
-    # Wait for next release to remove (current 1.12.4)
-    # apply_patch $my_loc/patches/global_planner.patch
-
-    #Patch Poco lib
-    apply_patch $my_loc/patches/poco.patch
-
-    # Plugin specific patches
-    if [ $use_pluginlib -ne 0 ]; then
-        # Patch pluginlib for static loading
-        apply_patch $my_loc/patches/pluginlib.patch
-        # apply_patch image_transport # to fix faulty export plugins
-        apply_patch $my_loc/patches/image_transport.patch
-    fi
```

Comment:
This special case is not handled anymore. In combination with [do_everything.sh:120ff](https://github.com/Intermodalics/ros_android/blob/bcccc2127b8e8e4ccb7e99a2a0bc9739755b269d/do_everything.sh#L120-L144) if `use_pluginlib` is disabled in [config.sh:12](https://github.com/Intermodalics/ros_android/blob/bcccc2127b8e8e4ccb7e99a2a0bc9739755b269d/config.sh#L12), applying the patches would break `pluginlib`.

The alternative would be to ignore package `pluginlib` and all its dependees when building with `use_pluginlib=0`, because the unpatched version would not work anyway on Android if I am not mistaken.

Reply:
AFAIU the unpatched version won't work; my rationale is that if you don't want `pluginlib`, applying a few extra packages doesn't add too much overhead, and you won't be able to use `pluginlib` anyway.

I agree that for correctness we should ignore `pluginlib` and deps, but on second thought I'd just drop the flag and build pluginlib support always, as it really doesn't add too much build time to the overall, and the flag just adds some more complexity to the scripts. WDYT?

Reply:
Always building with pluginlib support enabled would be fine for me.